demos/yelp_demo/demo_setup/README.md

```diff
@@ -412,123 +436,201 @@ ks param set --env minikube serving modelPath ${GCS_TRAINING_OUTPUT_DIR_LOCAL}/e
+USER_EMAIL=${CLUSTER}-user@${PROJECT}.iam.gserviceaccount.com
+USER_FILE=${HOME}/.ssh/${USER_EMAIL}.json
+
+gcloud iam service-accounts create ${CLUSTER}-admin --display-name=${CLUSTER}-admin
+gcloud iam service-accounts create ${CLUSTER}-user --display-name=${CLUSTER}-user
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${ADMIN_EMAIL} \
+  --role=roles/source.admin
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${ADMIN_EMAIL} \
+  --role=roles/servicemanagement.admin
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${ADMIN_EMAIL} \
+  --role=roles/compute.networkAdmin
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${ADMIN_EMAIL} \
+  --role=roles/storage.admin
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${USER_EMAIL} \
+  --role=roles/cloudbuild.builds.editor
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${USER_EMAIL} \
+  --role=roles/viewer
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${USER_EMAIL} \
+  --role=roles/source.admin
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${USER_EMAIL} \
+  --role=roles/storage.admin
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${USER_EMAIL} \
+  --role=roles/bigquery.admin
+gcloud projects add-iam-policy-binding ${PROJECT} \
+  --member=serviceAccount:${USER_EMAIL} \
+  --role=roles/dataflow.admin
+
+gcloud iam service-accounts keys create ${ADMIN_FILE} \
+  --project ${PROJECT} \
+  --iam-account ${ADMIN_EMAIL}
+gcloud iam service-accounts keys create ${USER_FILE} \
+  --project ${PROJECT} \
+  --iam-account ${USER_EMAIL}
+```
 
 #### Create the cluster with gcloud
 
+To create a cluster with auto-provisioning, run the following commands
+(estimated: 30 minutes):
+
 Follow the instructions
 [here](https://cloud.google.com/tpu/docs/kubernetes-engine-setup) to create a
-GKE cluster for use with TPUs and GPUs:
+GKE cluster for use with TPUs and node autoprovisiong (NAP) (estimated: 30
+minutes):
 
 ```
 gcloud beta container clusters create ${CLUSTER} \
   --project ${DEMO_PROJECT} \
   --zone ${ZONE} \
-  --accelerator type=nvidia-tesla-k80,count=2 \
-  --cluster-version 1.10.6-gke.2 \
+  --cluster-version 1.11 \
   --enable-ip-alias \
   --enable-tpu \
   --machine-type n1-highmem-8 \
+  --num-nodes=5 \
   --scopes cloud-platform,compute-rw,storage-rw \
   --verbosity error
-```
-
-#### OPTIONAL: Create the cluster with Deployment Manager instead (no TPUs)
-
-Create a config file:
 
-```
-cp gke/cluster-kubeflow-demo-base.yaml gke/cluster-${DEMO_PROJECT}.yaml
-```
-
-Modify any desired values, such as cluster name. Issue the following command, which uses Deployment Manager to create a GKE
-cluster:
+# scale down cluster to 3 (initial 5 is just to prevent master restarts due to upscaling)
```

Comment:
Did you consider that instead of creating everything manually to enable GKE beta features you do

kfctl.sh generate platform

And then modify the deployment manager configs to switch to the beta API for GKE and  set whatever fields need to be set.

Reply:
That's a cleaner approach, I like it

Reply:
Created #393 to modify setup instructions